frontend/public/components/storage-class-form.tsx

```diff
@@ -81,30 +81,18 @@ class StorageClassFormWithTranslation extends React.Component<
 
   storageTypes = {};
 
-  // Fetch Storage type provisioners from different operators
-  // For CSI - provisionerType: 'csi'
-  // For Defaults - provisionerType: 'others'
-  getExtensionsStorageClassProvisioners = (provisionerType = Provisioner.OTHERS) => {
-    const extensionCSIProvisioners: ExtensionSCProvisionerProp = _.reduce(
-      this.props.params,
-      (res, value) => {
-        const obj = value.properties.getStorageClassProvisioner || {};
-        if (obj) {
-          const key = provisionerType;
-          const keyValue = obj[provisionerType];
-          res[key] = keyValue;
-        }
-        return res;
-      },
-      {},
-    );
-
-    return extensionCSIProvisioners[provisionerType];
-  };
+  getExtensions = () =>
+    this.props.params?.forEach((item) => {
+      Object.entries(item?.properties?.[Provisioner.CSI] ?? {})?.forEach(
+        ([key, val]) => (this.CSIStorageTypes[key] = val),
+      );
+      Object.entries(item?.properties?.[Provisioner.OTHERS] ?? {})?.forEach(
+        ([key, val]) => (this.defaultStorageTypes[key] = val),
+      );
+    });
 
   // For 'csi' storage type
-  CSIStorageTypes = Object.freeze({
-    ...this.getExtensionsStorageClassProvisioners(Provisioner.CSI),
+  CSIStorageTypes = {
```

Comment:
Do you have an issue tracking the migration of these types to be declarative?

Reply:
The question would be for the default types do we really need to migrate them? If ever a default type is introduced which needs a custom component like the ceph storage class providers do then we can move them out to console-app. But these types right now do not require any component. Wdyt?

Reply:
Not sure what having a component or not has to do with this.

You're maintaining 2 different methods of contribution and 2 code paths. It's simply a matter for code maintenance. One way to do something is preferred over multiple. I'm simply suggesting the migration be done now or tracked in an issue.

Reply:
Creating a Jira Task to track moving these from here to console-app.